src/analysis/vector/qgsgeometrysnapper.cpp

```diff
@@ -100,6 +102,13 @@ bool QgsSnapIndex::SegmentSnapItem::getProjection( const QgsPoint &p, QgsPoint &
   return true;
 }
 
+bool QgsSnapIndex::SegmentSnapItem::withinDistance( const QgsPoint &p, const double tolerance )
+{
+  QgsLineString line( idxFrom->point(), idxTo->point() );
+  QgsGeos geos( &line );
```

Comment:
exporting a two point line string out to geos to calculate the point/line distance is quite inefficient. QgsGeometryUtils::sqrDistToLine is a better choice here.